>>>>>>> Task-1 Table Booking and Online Delivery

In [100]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

In [101]:
df1=pd.read_csv("Dataset.csv")
df1.head()

Restaurant ID         Restaurant Name  Country Code              City  \
0        6317637        Le Petit Souffle           162       Makati City   
1        6304287        Izakaya Kikufuji           162       Makati City   
2        6300002  Heat - Edsa Shangri-La           162  Mandaluyong City   
3        6318506                    Ooma           162  Mandaluyong City   
4        6314302             Sambo Kojin           162  Mandaluyong City   

                                             Address  \
0  Third Floor, Century City Mall, Kalayaan Avenu...   
1  Little Tokyo, 2277 Chino Roces Avenue, Legaspi...   
2  Edsa Shangri-La, 1 Garden Way, Ortigas, Mandal...   
3  Third Floor, Mega Fashion Hall, SM Megamall, O...   
4  Third Floor, Mega Atrium, SM Megamall, Ortigas...   

                                     Locality  \
0   Century City Mall, Poblacion, Makati City   
1  Little Tokyo, Legaspi Village, Makati City   
2  Edsa Shangri-La, Ortigas, Mandaluyong City   
3      SM Megamall, Ortigas, Mandaluyong City   
4      SM Megamall, Ortigas, Mandaluyong City   

                                    Locality Verbose   Longitude   Latitude  \
0  Century City Mall, Poblacion, Makati City, Mak...  121.027535  14.565443   
1  Little Tokyo, Legaspi Village, Makati City, Ma...  121.014101  14.553708   
2  Edsa Shangri-La, Ortigas, Mandaluyong City, Ma...  121.056831  14.581404   
3  SM Megamall, Ortigas, Mandaluyong City, Mandal...  121.056475  14.585318   
4  SM Megamall, Ortigas, Mandaluyong City, Mandal...  121.057508  14.584450   

                           Cuisines  ...          Currency Has Table booking  \
0        French, Japanese, Desserts  ...  Botswana Pula(P)               Yes   
1                          Japanese  ...  Botswana Pula(P)               Yes   
2  Seafood, Asian, Filipino, Indian  ...  Botswana Pula(P)               Yes   
3                   Japanese, Sushi  ...  Botswana Pula(P)                No   
4                  Japanese, Korean  ...  Botswana Pula(P)               Yes   

  Has Online delivery Is delivering now Switch to order menu Price range  \
0                  No                No                   No           3   
1                  No                No                   No           3   
2                  No                No                   No           4   
3                  No                No                   No           4   
4                  No                No                   No           4   

   Aggregate rating  Rating color Rating text Votes  
0               4.8    Dark Green   Excellent   314  
1               4.5    Dark Green   Excellent   591  
2               4.4         Green   Very Good   270  
3               4.9    Dark Green   Excellent   365  
4               4.8    Dark Green   Excellent   229  

[5 rows x 21 columns]

Create new features like "Has Table Booking"
or "Has Online Delivery" by encoding
categorical variable  (Also a part of TASK-3)s.

In [102]:
df1['Has Table booking'].replace({'Yes':1,'No':0},inplace=True)
df1['Has Online delivery'].replace({'Yes':1,'No':0},inplace=True)
df1['Is delivering now'].replace({'Yes':1,'No':0},inplace=True)
df1['Switch to order menu'].replace({'Yes':1,'No':0},inplace=True)


df1['Has Table booking'] = pd.to_numeric(df1['Has Table booking'])
df1['Has Online delivery'] = pd.to_numeric(df1['Has Online delivery'])
df1['Is delivering now'] = pd.to_numeric(df1['Is delivering now'])
df1['Switch to order menu'] = pd.to_numeric(df1['Switch to order menu'])

In [103]:
df1.head()


Restaurant ID         Restaurant Name  Country Code              City  \
0        6317637        Le Petit Souffle           162       Makati City   
1        6304287        Izakaya Kikufuji           162       Makati City   
2        6300002  Heat - Edsa Shangri-La           162  Mandaluyong City   
3        6318506                    Ooma           162  Mandaluyong City   
4        6314302             Sambo Kojin           162  Mandaluyong City   

                                             Address  \
0  Third Floor, Century City Mall, Kalayaan Avenu...   
1  Little Tokyo, 2277 Chino Roces Avenue, Legaspi...   
2  Edsa Shangri-La, 1 Garden Way, Ortigas, Mandal...   
3  Third Floor, Mega Fashion Hall, SM Megamall, O...   
4  Third Floor, Mega Atrium, SM Megamall, Ortigas...   

                                     Locality  \
0   Century City Mall, Poblacion, Makati City   
1  Little Tokyo, Legaspi Village, Makati City   
2  Edsa Shangri-La, Ortigas, Mandaluyong City   
3      SM Megamall, Ortigas, Mandaluyong City   
4      SM Megamall, Ortigas, Mandaluyong City   

                                    Locality Verbose   Longitude   Latitude  \
0  Century City Mall, Poblacion, Makati City, Mak...  121.027535  14.565443   
1  Little Tokyo, Legaspi Village, Makati City, Ma...  121.014101  14.553708   
2  Edsa Shangri-La, Ortigas, Mandaluyong City, Ma...  121.056831  14.581404   
3  SM Megamall, Ortigas, Mandaluyong City, Mandal...  121.056475  14.585318   
4  SM Megamall, Ortigas, Mandaluyong City, Mandal...  121.057508  14.584450   

                           Cuisines  ...          Currency Has Table booking  \
0        French, Japanese, Desserts  ...  Botswana Pula(P)                 1   
1                          Japanese  ...  Botswana Pula(P)                 1   
2  Seafood, Asian, Filipino, Indian  ...  Botswana Pula(P)                 1   
3                   Japanese, Sushi  ...  Botswana Pula(P)                 0   
4                  Japanese, Korean  ...  Botswana Pula(P)                 1   

   Has Online delivery  Is delivering now  Switch to order menu  Price range  \
0                    0                  0                     0            3   
1                    0                  0                     0            3   
2                    0                  0                     0            4   
3                    0                  0                     0            4   
4                    0                  0                     0            4   

   Aggregate rating  Rating color Rating text Votes  
0               4.8    Dark Green   Excellent   314  
1               4.5    Dark Green   Excellent   591  
2               4.4         Green   Very Good   270  
3               4.9    Dark Green   Excellent   365  
4               4.8    Dark Green   Excellent   229  

[5 rows x 21 columns]

In [104]:
number_of_tables_having_booking =sum(df1['Has Table booking']==1)
total_number_of_hotels =df1['Has Table booking'].count()

percentage_of_restaurants_offering_booking = (number_of_tables_having_booking*100)/(total_number_of_hotels)


In [105]:
percentage_of_restaurants_offering_booking

12.124384881164277

Hence the percenatage of restaurants offering table booking is 12.12%

In [106]:
#overall average rating
average_rating=df1['Aggregate rating'].mean()
average_rating

2.66637001361114

In [107]:
df2=df1.loc[df1['Has Table booking']==1]           #df2 has all rows with table booking

df3=df1.loc[df1['Has Table booking']==0]           #df3 has all rows without table booking


In [108]:
# average rating for having table booking

avg_rating = (df2['Aggregate rating']).mean()
avg_rating

3.4419689119170984

In [109]:
# average rating for having table booking

avg_rating_wo = (df3['Aggregate rating']).mean()
avg_rating_wo

2.559358989634219

Hence we see that the average rating is higher for restaurants with table booking (3.44) compared to restaurants without table booking (2.55)

In [110]:
df1['Price range'].unique()

array([3, 4, 2, 1], dtype=int64)

There are total 4 price ranges 1, 2, 3, and 4 now let's analyse the availiability of online delivery for each price range

In [111]:
#for price range 1
df4=df1.loc[df1['Price range']==1]

#for price range 2
df5=df1.loc[df1['Price range']==2]


#for price range 3
df6=df1.loc[df1['Price range']==3]


#for price range 4
df7=df1.loc[df1['Price range']==4]

In [112]:
total_numbers_of_restaurants_in_1=df4['Has Online delivery'].count()
numbers_of_restaurants_having_online_delivery_in_1=sum(df4['Has Online delivery']==1)
percentage_of_restaurants_having_online_delivery_in_1 = (numbers_of_restaurants_having_online_delivery_in_1*100)/total_numbers_of_restaurants_in_1




total_numbers_of_restaurants_in_2=df5['Has Online delivery'].count()
numbers_of_restaurants_having_online_delivery_in_2=sum(df5['Has Online delivery']==1)
percentage_of_restaurants_having_online_delivery_in_2 = (numbers_of_restaurants_having_online_delivery_in_2*100)/total_numbers_of_restaurants_in_2



total_numbers_of_restaurants_in_3=df6['Has Online delivery'].count()
numbers_of_restaurants_having_online_delivery_in_3=sum(df6['Has Online delivery']==1)
percentage_of_restaurants_having_online_delivery_in_3 = (numbers_of_restaurants_having_online_delivery_in_3*100)/total_numbers_of_restaurants_in_3





total_numbers_of_restaurants_in_4=df7['Has Online delivery'].count()
numbers_of_restaurants_having_online_delivery_in_4=sum(df7['Has Online delivery']==1)
percentage_of_restaurants_having_online_delivery_in_4 = (numbers_of_restaurants_having_online_delivery_in_4*100)/total_numbers_of_restaurants_in_4


In [113]:
percentage_of_restaurants_having_online_delivery_in_1

15.774077407740775

In [114]:
percentage_of_restaurants_having_online_delivery_in_2

41.31063283006746

In [115]:
percentage_of_restaurants_having_online_delivery_in_3

29.19034090909091

In [116]:
percentage_of_restaurants_having_online_delivery_in_4

9.044368600682594

The analysis of online delivery is given above for each price range. Online delivery is maximum for price range 2 and minimum for price range 4

>>>>>>>>TASK2-2 Price Range Analysis

In [117]:
print(df4.shape)
print(df5.shape)
print(df6.shape)
print(df7.shape)

(4444, 21)
(3113, 21)
(1408, 21)
(586, 21)


In [118]:
df1['Price range'].mode()

0    1
Name: Price range, dtype: int64

From the above statistical analysis we come to know that the price range 1 is the most common.

In [128]:
print('The avg aggregate rating is',df4['Aggregate rating'].mean())           #price range 1
print('The avg aggregate rating is',df5['Aggregate rating'].mean())           #price range 2
print('The avg aggregate rating is',df6['Aggregate rating'].mean())           #price range 3
print('The avg aggregate rating is',df7['Aggregate rating'].mean())           #price range 4

The avg aggregate rating is 1.999887488748875
The avg aggregate rating is 2.9410536460006425
The avg aggregate rating is 3.6833806818181816
The avg aggregate rating is 3.8179180887372017


The highest avg aggregate rating is in the price range 4 and the minimum avg aggrega

In [120]:
# for price range 1 the highest avg rating is 1.99~= 2 , so we have to find the corresponding rating color.
df8=df4.loc[df4['Aggregate rating']==2]
print(df8['Rating color'].unique())

['Red']


In [121]:
# for price range 2 the highest avg rating is 2.94~= 3 , so we have to find the corresponding rating color.
df9=df5.loc[df5['Aggregate rating']==3]
print(df9['Rating color'].unique())

['Orange']


In [122]:
# for price range 3 the highest avg rating is 3.68~=3.7 , so we have to find the corresponding rating color.
df10=df6.loc[df6['Aggregate rating']==3.7]
print(df10['Rating color'].unique())

['Yellow']


In [123]:
# for price range 4 the highest avg rating is 3.81 , so we have to find the corresponding rating color.
df11=df7.loc[df7['Aggregate rating']==3.8]
print(df11['Rating color'].unique())

['Yellow']


So from above analysis we found that for price range 1 ,2 ,3 and 4 the corresponding rating color are red , orange , yellow and yellow respectively

>>>>>>>>TASK-3 Feature Engineering

In [124]:
df1.head()

Restaurant ID         Restaurant Name  Country Code              City  \
0        6317637        Le Petit Souffle           162       Makati City   
1        6304287        Izakaya Kikufuji           162       Makati City   
2        6300002  Heat - Edsa Shangri-La           162  Mandaluyong City   
3        6318506                    Ooma           162  Mandaluyong City   
4        6314302             Sambo Kojin           162  Mandaluyong City   

                                             Address  \
0  Third Floor, Century City Mall, Kalayaan Avenu...   
1  Little Tokyo, 2277 Chino Roces Avenue, Legaspi...   
2  Edsa Shangri-La, 1 Garden Way, Ortigas, Mandal...   
3  Third Floor, Mega Fashion Hall, SM Megamall, O...   
4  Third Floor, Mega Atrium, SM Megamall, Ortigas...   

                                     Locality  \
0   Century City Mall, Poblacion, Makati City   
1  Little Tokyo, Legaspi Village, Makati City   
2  Edsa Shangri-La, Ortigas, Mandaluyong City   
3      SM Megamall, Ortigas, Mandaluyong City   
4      SM Megamall, Ortigas, Mandaluyong City   

                                    Locality Verbose   Longitude   Latitude  \
0  Century City Mall, Poblacion, Makati City, Mak...  121.027535  14.565443   
1  Little Tokyo, Legaspi Village, Makati City, Ma...  121.014101  14.553708   
2  Edsa Shangri-La, Ortigas, Mandaluyong City, Ma...  121.056831  14.581404   
3  SM Megamall, Ortigas, Mandaluyong City, Mandal...  121.056475  14.585318   
4  SM Megamall, Ortigas, Mandaluyong City, Mandal...  121.057508  14.584450   

                           Cuisines  ...          Currency Has Table booking  \
0        French, Japanese, Desserts  ...  Botswana Pula(P)                 1   
1                          Japanese  ...  Botswana Pula(P)                 1   
2  Seafood, Asian, Filipino, Indian  ...  Botswana Pula(P)                 1   
3                   Japanese, Sushi  ...  Botswana Pula(P)                 0   
4                  Japanese, Korean  ...  Botswana Pula(P)                 1   

   Has Online delivery  Is delivering now  Switch to order menu  Price range  \
0                    0                  0                     0            3   
1                    0                  0                     0            3   
2                    0                  0                     0            4   
3                    0                  0                     0            4   
4                    0                  0                     0            4   

   Aggregate rating  Rating color Rating text Votes  
0               4.8    Dark Green   Excellent   314  
1               4.5    Dark Green   Excellent   591  
2               4.4         Green   Very Good   270  
3               4.9    Dark Green   Excellent   365  
4               4.8    Dark Green   Excellent   229  

[5 rows x 21 columns]

In [125]:
df1['Address length']=df1['Address'].str.len()
df1['Restaurant Name length']=df1['Restaurant Name'].str.len()
df1.head()

Restaurant ID         Restaurant Name  Country Code              City  \
0        6317637        Le Petit Souffle           162       Makati City   
1        6304287        Izakaya Kikufuji           162       Makati City   
2        6300002  Heat - Edsa Shangri-La           162  Mandaluyong City   
3        6318506                    Ooma           162  Mandaluyong City   
4        6314302             Sambo Kojin           162  Mandaluyong City   

                                             Address  \
0  Third Floor, Century City Mall, Kalayaan Avenu...   
1  Little Tokyo, 2277 Chino Roces Avenue, Legaspi...   
2  Edsa Shangri-La, 1 Garden Way, Ortigas, Mandal...   
3  Third Floor, Mega Fashion Hall, SM Megamall, O...   
4  Third Floor, Mega Atrium, SM Megamall, Ortigas...   

                                     Locality  \
0   Century City Mall, Poblacion, Makati City   
1  Little Tokyo, Legaspi Village, Makati City   
2  Edsa Shangri-La, Ortigas, Mandaluyong City   
3      SM Megamall, Ortigas, Mandaluyong City   
4      SM Megamall, Ortigas, Mandaluyong City   

                                    Locality Verbose   Longitude   Latitude  \
0  Century City Mall, Poblacion, Makati City, Mak...  121.027535  14.565443   
1  Little Tokyo, Legaspi Village, Makati City, Ma...  121.014101  14.553708   
2  Edsa Shangri-La, Ortigas, Mandaluyong City, Ma...  121.056831  14.581404   
3  SM Megamall, Ortigas, Mandaluyong City, Mandal...  121.056475  14.585318   
4  SM Megamall, Ortigas, Mandaluyong City, Mandal...  121.057508  14.584450   

                           Cuisines  ...  Has Online delivery  \
0        French, Japanese, Desserts  ...                    0   
1                          Japanese  ...                    0   
2  Seafood, Asian, Filipino, Indian  ...                    0   
3                   Japanese, Sushi  ...                    0   
4                  Japanese, Korean  ...                    0   

  Is delivering now  Switch to order menu  Price range  Aggregate rating  \
0                 0                     0            3               4.8   
1                 0                     0            3               4.5   
2                 0                     0            4               4.4   
3                 0                     0            4               4.9   
4                 0                     0            4               4.8   

   Rating color  Rating text  Votes Address length Restaurant Name length  
0    Dark Green    Excellent    314             71                     16  
1    Dark Green    Excellent    591             67                     16  
2         Green    Very Good    270             56                     22  
3    Dark Green    Excellent    365             70                      4  
4    Dark Green    Excellent    229             64                     11  

[5 rows x 23 columns]

We have added two new columns in the df1 dataset to calculate the length of restaurant name and its address

The part of converting into categorical variables has been already done in in initial part of TASK-1